core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronAgent.java

```diff
@@ -0,0 +1,189 @@
+package org.openstack4j.openstack.networking.domain;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import org.openstack4j.model.network.Agent;
+import org.openstack4j.model.network.builder.AgentBuilder;
+import org.openstack4j.openstack.common.ListResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+import com.google.common.base.Objects;
+
+/**
+ * Neutron Agent
+ *
+ * @author Yin Zhang
+ */
+@JsonRootName("agent")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class NeutronAgent implements Agent {
+
+    public static class Agents extends ListResult<NeutronAgent> {
+
+        private static final long serialVersionUID = 1L;
+
+        @JsonProperty("agents")
+        private List<NeutronAgent> agents;
+
+        @Override
+        public List<NeutronAgent> value() {
+            return agents;
+        }
+    }
+
+    private static final long serialVersionUID = 1L;
+    private String binary;
+    private String description;
+    @JsonProperty("admin_state_up")
+    private boolean adminStateUp;
+    @JsonProperty("heartbeat_timestamp")
+    private String heartbeatTimeStamp;
+    private boolean alive;
+    private String topic;
+    private String host;
+    @JsonProperty("agent_type")
+    private Type agentType;
+    @JsonProperty("created_at")
+    private String createdAt;
+    @JsonProperty("started_at")
+    private String startedAt;
+
+    private String id;
+
+    private static final Logger LOG = LoggerFactory.getLogger(NeutronAgent.class);
+
+    /**
+     * local help method to parse a string with format "yyyy-MM-dd HH:mm:ss" into a Date
+     * @param str - string to be parsed
+     * @return Date
+     */
+    private Date parseDate(String str) {
```

Comment:
Just an idea: Could use https://github.com/ContainX/openstack4j/blob/master/core/src/main/java/org/openstack4j/openstack/internal/Parser.java instead of doing it here.
Makes sense?